engine/src/main/java/org/hibernate/validator/internal/cfg/context/TypeConstraintMappingContextImpl.java

```diff
@@ -133,8 +133,7 @@ public PropertyConstraintMappingContext property(String property, ElementType el
 
 	@Override
 	public PropertyConstraintMappingContext field(String property) {
-		Contracts.assertNotNull( property, "The property name must not be null." );
-		Contracts.assertNotEmpty( property, MESSAGES.propertyNameMustNotBeEmpty() );
+		Contracts.assertNeitherNullNorEmpty( property, MESSAGES.propertyNameMustBeNeitherNullNorEmpty() );
```

Comment:
Hum, sorry if I wasn't clear but I did expect you to only change the method content, not the method name or the message.

Being not empty implies being not null. See `NotEmptyValidatorForCharSequence` for instance :).